### ISSUE TITLE
Range diff: white text on yellow background

### ISSUE THREAD

**Issue by mstv** [gitextensions/gitextensions#11940]
### Environment

- Git Extensions v5.0.0
- Git 2.46.0.windows.1

### Issue description

RangeDiff seems to need a specific color configuration, too, in order to avoid white text on yellow background.

![grafik](https://github.com/user-attachments/assets/48b4d030-0d8c-4c54-8850-46037d39b081)

(Workaround: select the affected line. I have not searched for the regarding color slot yet.)

### Steps to reproduce

view range diff with change in metadata of commit

### Diagnostics

```
-c·color.ui=never·-c·diff.colorMovedWS=no·-c·diff.colorMoved=dimmed-zebra·-c·color.diff.old="red·reverse"·-c·color.diff.new="green·reverse"·-c·color.diff.oldMoved="black·brightmagenta"·-c·color.diff.newMoved="black·brightblue"·-c·color.diff.oldMovedAlternative="black·brightcyan"·-c·color.diff.newMovedAlternative="black·brightyellow"·-c·color.diff.oldMovedDimmed="magenta·dim·reverse"·-c·color.diff.newMovedDimmed="blue·dim·reverse"·-c·color.diff.oldMovedAlternativeDimmed="cyan·dim·reverse"·-c·color.diff.newMovedAlternativeDimmed="yellow·dim·reverse"·-c·color.diff.contextBold="normal·bold·reverse"·-c·color.diff.oldBold="brightred·reverse"·-c·color.diff.newBold="brightgreen··reverse"·range-diff·--find-renames·--find-copies·--color=always·--unified=<hash1>...<hash2>·--·"file.c"::c:\workdir\::False::
-------------------------------------

\x1b[7;31m1:··4059497e2·\x1b[m\x1b[33m!\x1b[m\x1b[7;32m·1:··<hash>\x1b[m\x1b[33m·Subject\x1b[m\n
····\x1b[7m\x1b[7;37m@@\x1b[m\n
······##·Metadata·##\x1b[m\n
····\x1b[7m\x1b[7;31m-\x1b[m\x1b[2mAuthor:·<author1>\x1b[m\n
····\x1b[7m\x1b[7;32m+\x1b[m\x1b[1;7mAuthor:·<author2>\x1b[m\n
·····\x1b[m\n
```

**Comment by gerhardol** [comment]
Have you configured the Git colors?
This should be overridden by the following:

-c·color.diff.newMovedAlternative="black·brightyellow"

**Comment by mstv** [comment]
> Have you configured the Git colors?

No, I have not. 
Otherwise, '-c·color.diff.newMovedAlternative="black·brightyellow"' would not have been in the command line, would it?
I only specifiy unrelated colors:
```
$ git config -l | grep color
color.ui=auto
color.grep.matchselected=yellow reverse
color.diff.frag=white reverse
color.diff.func=white reverse
```

**Comment by gerhardol** [comment]
> [1;7m

That is only bold,reverse not specifying yellow. The green color is 'reset' after the +.
bash: echo -e "\x1b[7m\x1b[7;32m+\x1b[m\x1b[1;7mAuthor:·<author2>\x1b[m\n"

![image](https://github.com/user-attachments/assets/4eb32a77-6dc4-4407-b88f-0161d1d6174f)

How is the yellow set?
(difftastic output is formatted, other should appear as in git)

**Comment by mstv** [comment]
The build actually is beyond 5.0.0. It includes the branch 'feature/difflineinfo-coloring'.
In 5.0.0, the entire line is black on green.

**Comment by gerhardol** [comment]
> In 5.0.0, the entire line is black on green.

Current foreground color is remembered too long in 5.0.0. The spec is not clear... I have not seen that make a difference with the Git output though.
But how is yellow rendered?

**Comment by mstv** [comment]
Bisected down to #11849.

STR: select the head of my branches "repro/rangediff-a" and "repro/rangediff-b" and open the range diff

**Comment by gerhardol** [comment]
I have no white text (or yellow background) for this range diff.
There are not so many slots in range-diff https://git-scm.com/docs/git-config#Documentation/git-config.txt-colordiffltslotgt
The command indicates that Git >= 2.26 is used (white foreground on bold background could occur in older Git, GE selects wrong foreground on your PC...).

The output from the Git command looks, there is no manipulation done then. No clue right now...

**Comment by mstv** [comment]
> No clue right now

me neither, same behavior at home and at work

- Git Extensions 33.33.33
- Build 211eaf1c1bd25ca349a95f9edd75a701994522fd
- Git 2.46.0.windows.1
- Microsoft Windows NT 10.0.19045.0
- .NET 8.0.8
- DPI 96dpi (no scaling)
- Portable: False

![image](https://github.com/user-attachments/assets/61b3ff05-6e48-450c-a991-f74f9f3bf0ab)

**Comment by RussKie** [comment]
Stumbled across the same today in the v5 release (5.0.0.17886, Git 2.46.1.windows.1):
![image](https://github.com/user-attachments/assets/342d06fe-c498-46af-a856-e98431568130)

```
Arguments:   -c color.ui=never -c diff.submodule=short -c diff.noprefix=false -c diff.mnemonicprefix=false -c diff.ignoreSubmodules=none -c core.safecrlf=false -c diff.colorMovedWS=no -c diff.wordRegex="[a-z0-9_]+|." -c diff.colorMoved=dimmed-zebra -c color.diff.old="red reverse" -c color.diff.new="green reverse" -c color.diff.oldMoved="magenta bold reverse" -c color.diff.newMoved="blue bold reverse" -c color.diff.oldMovedAlternative="cyan bold reverse" -c color.diff.newMovedAlternative="yellow bold reverse" -c color.diff.oldMovedDimmed="magenta dim reverse" -c color.diff.newMovedDimmed="blue dim reverse" -c color.diff.oldMovedAlternativeDimmed="cyan dim reverse" -c color.diff.newMovedAlternativeDimmed="yellow dim reverse" diff --find-renames --find-copies --color=always --ignore-all-space --unified=3 --cached "724c32e5c04e843575f7368f7a76a6de8f034cec" -- "file.cs"
```

**Comment by gerhardol** [comment]
> Stumbled across the same today in the v5 release (5.0.0.17886, Git 2.46.1.windows.1):
> ![image](https://github.com/user-attachments/assets/342d06fe-c498-46af-a856-e98431568130)
> 
> ```
> Arguments:   -c color.ui=never -c diff.submodule=short -c diff.noprefix=false -c diff.mnemonicprefix=false -c diff.ignoreSubmodules=none -c core.safecrlf=false -c diff.colorMovedWS=no -c diff.wordRegex="[a-z0-9_]+|." -c diff.colorMoved=dimmed-zebra -c color.diff.old="red reverse" -c color.diff.new="green reverse" -c color.diff.oldMoved="magenta bold reverse" -c color.diff.newMoved="blue bold reverse" -c color.diff.oldMovedAlternative="cyan bold reverse" -c color.diff.newMovedAlternative="yellow bold reverse" -c color.diff.oldMovedDimmed="magenta dim reverse" -c color.diff.newMovedDimmed="blue dim reverse" -c color.diff.oldMovedAlternativeDimmed="cyan dim reverse" -c color.diff.newMovedAlternativeDimmed="yellow dim reverse" diff --find-renames --find-copies --color=always --ignore-all-space --unified=3 --cached "724c32e5c04e843575f7368f7a76a6de8f034cec" -- "file.cs"
> ```
> 

> color.diff.newMovedAlternative="yellow bold reverse”

That should only be set if Git is older than 2.26 if the boldyellow fix is in 5.0
I did not dare to change the heuristics that selects forecolor...

**Comment by gerhardol** [comment]
@RussKie Your issue is something else, not handled by #11962 
The command line indicates the Git version is <2.26, what is the actual version?
